### PR TITLE
Api can provide list of rental spaces

### PR DESCRIPTION
**Pull request by Cma-B**
### User Story
```
As an api
In order to display the list of rental spaces to a client
I would like to provide an end point for spaces index

```
## Changes proposed in this pull request:
* define index action for the listing 
* adds serializers for listing index
* adds listing model 
## What I have learned working on this feature:
* working more with serializers
## Packages/Gems added
-  Active model serializers

### CODE REVIEW

File: app/models/listing.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class Listing < ApplicationRecord
+
```
Comment (viktorsommar) [inline app/models/listing.rb]: No validations?

File: db/migrate/20200831121034_create_listings.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class CreateListings < ActiveRecord::Migration[6.0]
+  def change
+    create_table :listings do |t|
+      t.string :category
+      t.text :description
+      t.string :scene
```
Comment (viktorsommar) [inline db/migrate/20200831121034_create_listings.rb]: We need a name or title or something

File: app/serializers/listing_index_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+class ListingIndexSerializer < ActiveModel::Serializer
+  include Rails.application.routes.url_helpers
+  attributes :id, :category, :description, :scene, :height, :address, :price
```
Comment (Alex-Bellamy) [inline app/serializers/listing_index_serializer.rb]: title of listing or equivalent required for 'pitching' your listing

File: spec/factories/listings.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :listing do
+    category {'Car park'}
+    description {'vacant'}
+    scene {'indoors'}
```
Comment (viktorsommar) [inline spec/factories/listings.rb]: Name or title?

File: spec/models/listing_spec.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+RSpec.describe Listing, type: :model do
+
+  it 'should have a valid Factory' do
+    expect(create(:listing)).to be_valid
+  end
+
+  describe 'Database table' do
+    it { is_expected.to have_db_column :category }
+    it { is_expected.to have_db_column :description }
```
Comment (viktorsommar) [inline spec/models/listing_spec.rb]: Name or title?

File: spec/requests/api/v1/listings/index_spec.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+    it 'should return a 200 response' do
+    expect(response).to have_http_status 200
+    end
+
+    it 'should return listings' do
+    expect(response_json["listings"].count).to eq 3
+    end
+  end 
+
```
Comment (viktorsommar) [inline spec/requests/api/v1/listings/index_spec.rb:25]: Can we have a test to check the lead of one listing and the adress of another? For example

**Review by viktorsommar — CHANGES_REQUESTED** [review]

File: spec/requests/api/v1/listings/index_spec.rb (hunk 6 above)
Comment (Alex-Bellamy) [inline spec/requests/api/v1/listings/index_spec.rb:25]: validation specs?

File: app/serializers/listing_index_serializer.rb (hunk 3 above)
Comment (Cma-B) [reply app/serializers/listing_index_serializer.rb]: we changed that to just Id, category and lead plus the Scene that shows if the car park is inside or outside.

File: app/models/listing.rb (hunk 1 above)
Comment (Cma-B) [reply app/models/listing.rb]: the validations has been added!

File: db/migrate/20200831121034_create_listings.rb (hunk 2 above)
Comment (Cma-B) [reply db/migrate/20200831121034_create_listings.rb]: Lead is added for this purpose!

**Review by felixoppongwillborg — APPROVED** [review]

File: db/migrate/20200831121034_create_listings.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class CreateListings < ActiveRecord::Migration[6.0]
+  def change
+    create_table :listings do |t|
+      t.string :category
+      t.text :lead
```
Comment (viktorsommar) [inline db/migrate/20200831121034_create_listings.rb]: doesn't really matter but it could have been a string for the lead

**Review by viktorsommar — APPROVED** [review]

File: db/migrate/20200831121034_create_listings.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+      t.string :scene
```
Comment (oliverochman) [inline db/migrate/20200831121034_create_listings.rb]: perhaps turn this into an enum

**Review by oliverochman — APPROVED** [review]

**Review by felixoppongwillborg — APPROVED** [review]

**Review by viktorsommar — APPROVED** [review]

**Review by Alex-Bellamy — APPROVED** [review]

**Pull request merged**